### PR TITLE
Add glossary item for "pull"

### PR DESCRIPTION
**Pull request by todaywasawesome**
From this discussion: https://github.com/open-gitops/project/discussions/134

> In light of ongoing discussions and questions in slack, of which this is the [most recent example](https://cloud-native.slack.com/archives/C01G9DEE88M/p1671020692512439), it seems prudent to explain what pull means.
> 
> Sometimes, readers understand "pull" to mean that GitOps agents must live in the infrastructure under management so that it is self-sufficient and does not require external agents. This however, is not the intention of the word pull. The word pull here is specifically in reference to how desired state is discovered.
> 

Signed-off-by: Dan Garfield <dan@codefresh.io>

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

### CODE REVIEW

**Comment by ubaumann** [comment]
I would like it if there would be something about the fact, that the GitOps agent needs to be able to access the SSoT at any time.

Maybe we add the following sentence?

_In contrast to traditional, CI/CD pipelines which "push" event based triggers to discover state changes, in GitOps desired state is "pulled"._ **The GitOps agents need to be able to access the desired state at any time to avoid drifts.** _This state may be reconciled with local or remote actual states. In GitOps agents are expected to pull both desired and actual states without the requirement of events. Webhooks may be used to speed up this process, but should a webhook fail, the agent is still expected to discover changes to desired state by pulling from the source of truth._

**Comment by scottrigby** [comment]
Thanks @todaywasawesome for hopping on this.

@ubaumann I agree, from Slack and elaborated in this discussion answer https://github.com/open-gitops/project/discussions/134#discussioncomment-4427444 WDYT?

File: GLOSSARY.md
Comment (scottrigby) [inline GLOSSARY.md:24]: How about moving some of the language we already have under the "Reconciliation" glossary item? IMO this part fits better under "Pull" - we should remove this part:

> Contrary to traditional CI/CD where automation is generally driven by pre-set triggers, in GitOps reconciliation is triggered whenever there is a divergence. Divergence could be due to the actual state unintentionally [drifting](https://github.com/open-gitops/documents/blob/release-v1.0.0/GLOSSARY.md#drift) from the desired state declarations, or a new desired state declaration version having been changed intentionally.

And add to something that explains this clearly within the context of Principle 3, like:

```suggestion
    [Principle 3](./PRINCIPLES.md) specifies the desired state must be "pulled" rather than "pushed", primarily because the software agents must be able to access the [desired state](#desired-state) from the [state store](#state-store) at _any_ time, not only when there is an intentional change in the state store triggering a push event.
    This is a prerequisite for [reconciliation](#reconciliation) to happen [continuously](#continuous), as specified in [principle 4](./PRINCIPLES.md).
    Note that – in contrast to traditional CI/CD, where automation is generally driven by pre-set triggers – in GitOps, [reconciliation](#reconciliation) is triggered _whenever_ there is a divergence.
    Divergence could be due to the actual state unintentionally [drifting](#drift) from the desired state declarations – not only due to a new desired state declaration version having been changed intentionally.
```

**Review by scottrigby — COMMENTED** [review]
Dan what do you think about this? building on what you wrote but also with more direct ties to existing content. If so you can merge this suggestion into your PR branch directly (number 3 in [Incorporating feedback in your pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request)).

**Comment by scottrigby** [comment]
Great 👍 

Now just looks like DCO bot is unhappy with email mismatch:

> Commit sha: 984ed8e, Author: Dan Garfield, Committer: GitHub; Can not find "Dan Garfield dan@todaywasawesome.com", in ["Dan Garfield dan@codefresh.io", "Dan Garfield dan@codefresh.io"].

Steps to fix: https://github.com/open-gitops/documents/runs/12166015365

**Comment by ubaumann** [comment]
@todaywasawesome Looks good to me! It makes it much more straightforward. Thanks

**Review by scottrigby — APPROVED** [review]

**Comment by scottrigby** [comment]
After merging this, we can create a v1.1.0 draft similar to how we did 1.0.0.

**Review by niklasmtj — APPROVED** [review]
LGTM

**Review by christianh814 — APPROVED** [review]

**Review by roberthstrand — APPROVED** [review]

**Comment by scottrigby** [comment]
Ok this has been open for review in its current state for a good while now, has multiple collaborators/co-authors, and 4 maintainer approvals. Merging. Thanks everyone! 🚀

**Pull request merged**
